sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -1130,10 +1133,6 @@ private[hive] object HiveClientImpl extends Logging {
     Option(hc.getComment).map(field.withComment).getOrElse(field)
   }
 
-  private def verifyColumnDataType(schema: StructType): Unit = {
```

Comment:
This is a misleading step for the write path, the `schema` here is both produced and verified by Spark itself but reports a CANNOT_RECOGNIZE_HIVE_TYPE error to us

Reply:
After this change by quoting, this check is always positive

Reply:
It's better to keep this check since it doesn't hurt.

One of the side effects of this method is it will trigger the fallback code path here: https://github.com/apache/spark/blob/05728e4ff64e6684d7c6501f8a079e3b9aded9ed/sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala#L283-L289

This would ensure even if we generate a bad schema (e.g., a bug in `catalogString`), we will still write a schema that's working in Spark. Of cause, the fallback code path will not work for Hive. However, the Hive compatibility is the best effort and the Spark compatibility is more important.

Reply:
@yaooqinn can you answer this question? It can capture future bugs if Spark can't pass this roundtrip parsing test.

Reply:
OK, let me state it again. the Hive parser (in HMS) and the Spark SQL parser do the nested struct type parsing in completely opposite ways for quoting.

It's nonsense here to use the Spark SQL parser to verify schemas produced by itself but claim it as an HIVE_TYPE error.

For the callers, like create-table, alter-table commands that rely on the errors to re-store schemas into the spark properties, are still work correctly. Or perhaps, more correctly with the actual hive error.

Reply:
Based on the current work throughout the PR, Spark Parser will only perform additional checks on the schema stored in the HMS. This process is sufficient to identify incompatibilities when reading the Hive schema. For the writing process, it makes no sense for Spark to calibrate itself, a process that Hive's parser will help us with.